Tasks/PublishPipelineArtifactV1/task.loc.json

```diff
@@ -38,6 +38,48 @@
       "defaultValue": "",
       "required": false,
       "helpMarkDown": "ms-resource:loc.input.help.artifactName"
+    },
+    {
+      "name": "artifactType",
```

Comment:
If we are modifying the task fields, it has to be a major version update.

Reply:
hmm I'll discuss with Will about the major version change.

Reply:
If this version of the task only supports two options (and where one is super advanced), can we just have an optional boolean input like "Publish to file share"? If set (i.e. checked ON or set to "true"), the `fileSharePath` input becomes enabled. Default is false.

Reply:
Not sure what's the need of having a boolean field here. The current UI is like the build artifact task where by default it's using pipeline type unless the fileShareType is selected.

Reply:
The current updates shouldn't fail the existing builds. The bigger discussion is - should we create another major version because we're adding the support for UNC path?

Reply:
chrispat - we are adding UNC support to the Publish Pipeline Artifact V1 task, but these changes do not impact existing pipelines (i.e. non-breaking). There are associated agent changes which will require updating `minimumAgentVersion` (@jahsu-MSFT - correct?), which could cause technically cause a pipeline using a private agent to stop working (since the pipeline now requires a later agent).

Net-net: for this reason (or for any other --- i.e. prior art), would it be better to introduce a new major version (V2) instead?

Reply:
@jahquin-msft - yah, sorry .. I shouldn't have mixed different topics in the same thread, but since I already have .... :)

The reason for having a boolean `Publish to a file share` input (instead of what you have currently) is to simplify the UX for the common case (the vast majority of users won't be publishing to a file share). A simple and specific checkbox is better than a drop-down with two options, especially when the default option isn't actually that obvious (i.e. *"What does it mean to publish to Azure Pipelines?!"*).

Reply:
@willsmythe I think you meant to tag me? :) Still I think the current UI makes sense, or maybe I misunderstood your concept. Currently a user can only select "either pipeline type or fileshare type" (only one of them) so having an option to select either "pipeline" or "fileshare" makes sense. If the name is not clear, we can update the naming I suppose.

Reply:
Tag @chrispat  for the question regarding major version update.